viser/client/src/SceneTree.tsx

```diff
@@ -245,12 +255,45 @@ export function SceneNodeThreeObject(props: {
     <>
       {React.useMemo(() => makeObject(setRef), [makeObject, setRef])}
       {obj !== null && (
-        <SceneNodeThreeChildren
-          name={props.name}
-          useSceneTree={props.useSceneTree}
-          parent={obj}
-        />
+        <group>
+          <SceneNodeLabel 
+            visible={visibility && labelVisibility}
+            text={props.name} 
+          />
+          <SceneNodeThreeChildren
+            name={props.name}
+            useSceneTree={props.useSceneTree}
+            parent={obj}
+          />
+        </group>
       )}
     </>
   );
 }
+
+export type SceneNodeLabelProps = {
+  text: string;
+  visible?: boolean;
+}
+
+export function SceneNodeLabel({ text, visible }: SceneNodeLabelProps) {
+  if (!visible) {
+    return null;
+  }
+  if (text.trim() === "") { // Do not render label for object without name, e.g. root node
```

Comment:
maybe this condition could be merged with above?

Reply:
Fixed!